--- /root/copy/6-1/fft/fft.cpp	2010-03-30 18:00:34.000000000 +0000
+++ fft.cpp	2018-07-05 11:57:54.888088750 +0000
@@ -32,14 +32,14 @@
         case 1:
             gws[0] = x;
             gws[1] = 1;
-            lws[0] = 1;
-            lws[1] = 1;
+            lws[0] = 16;
+            lws[1] = 16;
             break;
         default:
             gws[0] = x;
             gws[1] = y;
-            lws[0] = 1;
-            lws[1] = 1;
+            lws[0] = 16;
+            lws[1] = 16;
             break;
     }
 
